subworkflows/local/vcf_impute_beagle5/main.nf

```diff
@@ -0,0 +1,71 @@
+include { BEAGLE5_BEAGLE                           } from '../../../modules/nf-core/beagle5/beagle'
+include { BCFTOOLS_INDEX as BCFTOOLS_INDEX_BEAGLE } from '../../../modules/nf-core/bcftools/index'
+include { BCFTOOLS_VIEW                           } from '../../../modules/nf-core/bcftools/view'
+
+
+workflow VCF_IMPUTE_BEAGLE5 {
+    take:
+    ch_input  // channel: [ [id, chr], vcf, tbi ]
+    ch_panel  // channel: [ [id, chr], vcf, tbi ]  
+    ch_map    // channel: [ [chr], map]
+
+    main:
+    ch_versions = Channel.empty()
+
+    // Branch input files based on format 
+    ch_input
+        .branch { meta, vcf, tbi ->
+            bcf: vcf.toString().contains('.bcf')
+            vcf: vcf.toString().contains('.vcf')
+        }
+        .set { ch_input_branched }
+
+    // Convert BCF to VCF if necessary
+    BCFTOOLS_VIEW(
+        ch_input_branched.bcf,
+        [],
+        [],
+        []
+    )
+    ch_versions = ch_versions.mix(BCFTOOLS_VIEW.out.versions.first())
+
+    // Combine VCF files 
+    ch_ready_vcf = ch_input_branched.vcf
+        .mix(BCFTOOLS_VIEW.out.vcf.join(BCFTOOLS_VIEW.out.csi))
+
+    // Prepare input channels for BEAGLE5 by combining VCF, panel, and map files
+    ch_beagle_input = ch_ready_vcf
+        .map { meta, vcf, tbi -> [meta.chr, meta, vcf, tbi] }
+        .combine(ch_panel.map { meta, vcf, idx -> [meta.chr, meta.id, vcf] }, by: 0)
+        .combine(ch_map.map { meta, map -> [meta.chr, map] }, by: 0) 
+        .multiMap { chr, target_meta, vcf, tbi, panel_id, panel_vcf, map ->
+            input: [target_meta, vcf] 
+            panel: panel_vcf
+            map: map
+        }
+
+    // Run BEAGLE5 imputation
+    BEAGLE5_BEAGLE(
+        ch_beagle_input.input,
+        ch_beagle_input.panel,
+        ch_beagle_input.map,
+        [],
+        []
+    )
```

Comment:
If you provide the channel as such they won't stay in sync.
You need all the input files to be in the same tuple.